src/accessibility/outputs.js

```diff
@@ -0,0 +1,418 @@
+    }
+  } else {
+    return;
+  }
+};
+
+/**
+ * <code class="language-javascript">gridOutput()</code>, formerly called
+ * table output, lays out the content of the canvas in the form of a grid
+ * (html table) based on the spatial location of each shape. A brief
+ * description of the canvas is available before the table output.
+ * This description includes: color of the background, size of the canvas,
+ * number of objects, and object types (example: "lavender blue canvas is
+ * 200 by 200 and contains 4 objects - 3 ellipses 1 rectangle"). The grid
+ * describes the content spatially, each element is placed on a cell of the
+ * table depending on its position. Within each cell an element the color
+ * and type of shape of that element are available (example: "orange ellipse").
+ * These descriptions can be selected individually to get more details.
+ * A list of elements where shape, color, location, and area are described
+ * (example: "orange ellipse location=top left area=1%") is also available.
+ *
+ * @method gridOutput
+ * @param  {Constant} [display] either LABEL or FALLBACK (Optional)
+ *
+ * @example
+ * //<div>
+ * //<code>
+ * //textOutput();
+ * //background(148, 196, 0);
+ * //fill(255, 0, 0);
+ * //ellipse(20, 20, 20, 20);
+ * //fill(0, 0, 255);
+ * //rect(50, 50, 50, 50);
+ * //</code>
+ * //</div>
+ *
+ *
+ * //<div>
+ * //<code>
+ * //let x = 0;
+ * //function draw() {
+ *   //textOutput();
+ *   //background(148, 196, 0);
+ *   //fill(255, 0, 0);
+ *   //ellipse(x, 20, 20, 20);
+ *   //fill(0, 0, 255);
+ *   //rect(50, 50, 50, 50);
+ *   //ellipse(20, 20, 20, 20);
+ *   //x += 0.1;
+ * //}
+ *
+ */
+
+p5.prototype.gridOutput = function(display) {
+  p5._validateParameters('gridOutput', arguments);
+  if (txtOut === false) {
+    grOut = true;
+    if (this.canvas !== undefined) {
+      this._setDefaults();
+      this._createOutput('grOut');
+    } else {
+      throw new Error('gridOutput() should be called after canvas is created');
+    }
+  } else {
+    return;
+  }
+};
+
+//helper function returns true when accessible outputs are true
+p5.prototype._addAccsOutput = function() {
+  if (txtOut === true || grOut === true) {
+    return true;
+  } else {
+    return false;
+  }
+};
+
+//helper function that creates html structure for accessible outputs
+p5.prototype._createOutput = function(type) {
+  let cnvId = this.canvas.id;
+  let cIdT = cnvId + type;
+  if (document.getElementById(cIdT) === null) {
+    document
+      .getElementById(cnvId)
+      .insertAdjacentHTML(
+        'afterend',
+        '<section id="' + cIdT + '" class="accessibleOutput"></section>'
+      );
+    if (type === 'txtOut') {
+      let inner = this._createTextOutput(cIdT);
+      document.getElementById(cIdT).innerHTML = inner;
+    }
+    if (type === 'grOut') {
+      let inner = this._createGridOutput(cIdT);
+      document.getElementById(cIdT).innerHTML = inner;
```

Comment:
I wonder if it's more performant to store key DOM elements internally rather than needing to use `getElementById` lookups for each update?